src/pages/organize/[orgId]/campaigns/index.tsx

```diff
@@ -33,14 +79,275 @@ export const getServerSideProps : GetServerSideProps = scaffold(async (ctx) => {
     }
 }, scaffoldOptions);
 
+const useStyles = makeStyles((theme) => ({
+    expandButton: {
+        textDecoration: 'underline',
+        textTransform: 'none',
+    },
+    speedDial: {
+        bottom: theme.spacing(4),
+        position: 'fixed',
+        right: theme.spacing(4),
+    },
+}));
+
 type AllCampaignsSummaryPageProps = {
     orgId: string;
 };
 
-const AllCampaignsSummaryPage: PageWithLayout<AllCampaignsSummaryPageProps> = () => {
+const AllCampaignsSummaryPage: PageWithLayout<AllCampaignsSummaryPageProps> = ({ orgId }) => {
+    const router = useRouter();
+    const classes = useStyles();
+    const intl = useIntl();
+    const campaignsQuery = useQuery(['campaigns', orgId], getCampaigns(orgId));
+    const upcomingEventsQuery = useQuery(['upcomingEvents', orgId], getUpcomingEvents(orgId));
+    const eventsQuery = useQuery(['events', orgId], getEvents(orgId));
+    const surveysQuery = useQuery(['surveys', orgId], getSurveys(orgId));
+    const callsQuery = useQuery(['calls', orgId], getAllCallAssignments(orgId));
+    const canvassesQuery = useQuery(['canvasses', orgId], getCanvasses(orgId));
+
+    const eventMutation = useMutation(postEvent(orgId));
+    const campaignMutation = useMutation(postCampaign(orgId));
+
+    const [expandedList, setExpandedList] = useState(false);
+    const [speedDialOpen, setSpeedDialOpen] = useState(false);
+    const [dialogOpen, setDialogOpen] = useState<null | string>(null);
+    const [filtered, setfiltered] = useState({
+        callAssignments: false,
+        canvasses: false,
+        standalones: false,
+        surveys: false,
+    });
+    const [cardHeight, setCardHeight] = useState(0);
+    const card = useRef<HTMLDivElement>(null);
+
+    useEffect(() => {
+        setCardHeight(card.current?.offsetHeight || 0);
+    }, []);
+
+    const campaigns = campaignsQuery.data || [];
+
+    const upcomingEvents = upcomingEventsQuery.data || [];
+    const events = eventsQuery.data || [];
+    const surveys = surveysQuery.data?.map(s => ({ ...s, key: `survey-${s.id}` })) || [];
+    const callAssignments = callsQuery.data?.map(c => ({ ...c, key: `call-${c.id}` })) || [];
+    const canvasses = canvassesQuery.data?.map(c => ({ ...c, key: `canvass-${c.id}` })) || [];
+    const standalones = events.filter(e => !e.campaign.id).map(s => ({ ...s, key: `standalone-${s.id}` }));
+
+    let unsorted = [
+        ... filtered.surveys ? [...surveys] : [],
+        ... filtered.canvasses ? [...canvasses] : [],
+        ... filtered.callAssignments ? [...callAssignments] : [],
+        ... filtered.standalones ? [...standalones] : [],
+    ];
+
+    if (Object.values(filtered).every(v => !v)) {
+        unsorted = [...surveys, ...canvasses, ...callAssignments, ...standalones];
```

Comment:
I would recommend you to not just use the regular objects from the API, but some sort of wrapper/envelope to indicate the type, e.g.: 

```tsx
unsorted = [
  ...surveys.map(survey => ({ type: 'survey', data: survey })),
  ...callAssignments.map(asmt => ({ type: 'call_assignment', data: asmt })),
  // etc
]
```

The reason for this is so that you can tell while rendering what kind of data you're working with, as they might need to be rendered slightly differently and link to different URLs.

This is not critical right now though, as we decided to hide this feature altogether anyway.